### ISSUE TITLE
Android - Steps Screen - Why are there Checkboxes in the Topics Section

### ISSUE THREAD

**Issue by jfc3-dol** [USDepartmentofLabor/Comply-Chain#33]
Why are there checkboxes in front of the “Topics” section items/links for each Step?

After checking with TalkBack, they are read as “white squares” which my guess is a type of bullet item. After going to a given screen, they are now checkmarks, and TalkBack doesn’t read anything for them.

Please see the attached screenshot.

![android - steps screen - checkboxes and checkmarks](https://user-images.githubusercontent.com/10423487/57155620-f3e0b000-6da9-11e9-9712-de1f749c6d8d.png)

**Comment by KyleLivingstonDOL** [comment]
Thanks for this John. The Checkboxes are a direct feedback from users wanting a way to indicate that they have finished those specific topics within a step. Once a topic is opened, the boxes are replaced with checkmarks.

I can see if our contractors can change the talk-back feature for those bullets to read a bit more specific. Otherwise, changing this feature would change the base-functionality of the app and how our users have asked us to update it.

**Comment by jfc3-dol** [comment]
Yes, improving the text for TalkBack that a given item is checked or not checked would be helpful to users. The key is that TalkBack is told which item is checked or not checked.

An example would be "Checkbox checked for Topic 2 - XYZ" or "Checkbox not checked for Topic 2 - XYZ". Doing it this way tells the user the state of the checkbox and which item it is.

**Comment by KyleLivingstonDOL** [comment]
Thanks John, we will do this.

**Comment by KyleLivingstonDOL** [comment]
Hi John,

Can we please close this item out? It is very similar to #34, which I will address.

Thanks,
Kyle

**Comment by jfc3-dol** [comment]
Kyle,

One issue is about the TalkBack not reading the check boxes correctly (#33) while the other is not letting a TalkBack user activate what are links but are not told they are (#34). So they are two separate issues that can be fixed at the same time. So I'm going to keep them open until I can test them.

**Comment by jfc3-dol** [comment]
TalkBackis reading the what looks like a checkbox as “white squares” and nothing for the one that is a checkmark. TalkBack should read is as "checkmark" or "Empty checkbox".

**Comment by KyleLivingstonDOL** [comment]
Hi John,

The square boxes we used are just that, square font boxes. They are like any other font (A, 1, Z, @, etc). Since they are not bullet points, Talkback is reading them correctly as white squares. As such, currently technology is not available to change how Talkback is reading these white squares. Therefore, we would like to put this item on hold until we can find the technology to program this.

Best,
Kyle

**Comment by jfc3-dol** [comment]
They are confusing to the screen reader user when they are read  “white square” or get nothing for the one with the checkmark. Instead couldn't you make the description text be "Unread" and "Read" or "Not viewed" or "Viewed" or something like that, if that is what those are supposed to represent/indicate?

**Comment by LeHaine** [comment]
The issue here is that we are using custom bullets that are characters. This is all done via CSS so everything aligns correctly as if we were doing a normal list. We set the new bullet via the content property which unfortunately has no way of having some sort of alternate text or label that can be read via screen readers.

**Comment by jfc3-dol** [comment]
Then why not use an actual checkbox, so the user will be told it's a checkbox that is check or not or make it an image with proper ALT text to tell the user it's "Unread" and "Read"?

**Comment by KyleLivingstonDOL** [comment]
> Then why not use an actual checkbox, so the user will be told it's a checkbox that is check or not or make it an image with proper ALT text to tell the user it's "Unread" and "Read"?

Hi John, I believe we looked into using an actual checkbox and there were none available to us that met our users' needs. ZTI come up with this solution that would best fit what our users' want when using Comply Chain. We designed most of the app (since most of the app is built around these 8 steps) with this in mind, including the timeline. As such, We are unable to scrap this at this point. When we are able to, we can go in and create alternative text (when that technology is available) so a screen reader is able to correctly read that information. I recommend that we table this until the technology allows us to do that.

-Kyle

**Comment by jfc3-dol** [comment]
If that is the case with this issue or any other that is not left open and not accessible OPA sends out a document listing the number of issues found, still open, and requests at least an e-mail stating the given agency is willing to take the risk if someone complains in whatever way that it's not accessible. Which has been done by all of them as far as I know.

**Comment by KyleLivingstonDOL** [comment]
Ok. Thanks John. I appreciate this information. Do you know what the procedure is to start this process?

**Comment by jfc3-dol** [comment]
Once we are done testing the entire Comply Chian application (iOS, Android, and web) I create and send to you and my boss OPAs findings of the number of issues found; the number closed, etc. At which point we then look for a signature (can be electronic or an e-mail) from the agency that they are willing to take the risk of having known accessibility issues when moving everything to PROD.

By documenting things in GitHub with all these types of comments, we could explain to someone that some of them are platform issues, etc. OPA doesn't want to put an agency or DOL at risk for complaints, etc. from people along with we are working on them.

**Comment by jfc3-dol** [comment]
Marking this issue for a "Future release" by using the label "future release" so we all know we are most likely not finishing it this release unless we manage to figure out how to do it.

**Comment by jfc3-dol** [comment]
TalkBack is till reading the empty checkbox as "white square", and the items that were reviewed or are checked makes a busy or noise instead of saying either "Reviewed" or "Not Reviewed" or similar such as "Read" or "Not read".

**Comment by jfc3-dol** [comment]
@pbhatt17 

For both iOS and Android, the screen readers (TalkBack and VoiceOver) are now at least reading "Reviewed" and "Not Reviewed" for the checkboxes.

So, I'm going to close this issue.
